### PR TITLE
Data Store new Actions

### LINKED ISSUE #2746: [ACTION] Data Store Actions
**Is there a specific app this action is for?**
`data_stores`

 ### New actions
- List keys in Data Store: `await this.db.keys()`
- Delete all keys in a Data Store: `await this.db.clear()` (include a note about this being an irreversible change, so use with caution)
- Check existence of keys: `await this.db.has(key)` (returns true/false)

### Updates to existing actions
- Can we use async options now to fetch existing keys for the `add-update-record` action?
- Update `delete-record` action to use `await this.db.delete(key)` (and rename action "Delete single record"

### PR DESCRIPTION
**Pull request by vellames-turing**
Closes #2746

### CODE REVIEW

File: components/data_stores/actions/delete-single-record/delete-single-record.mjs [hunk 1]
```diff
@@ -1,10 +1,10 @@
 import app from "../../data_stores.app.mjs";
 
 export default {
-  key: "data_stores-delete-record",
+  key: "data_stores-delete-single-record",
   name: "Delete a single record",
   description: "Delete a single record in your [Pipedream Data Store](https://pipedream.com/data-stores/).",
-  version: "0.0.4",
+  version: "0.0.2",
```
Comment (michelle0927) [inline components/data_stores/actions/delete-single-record/delete-single-record.mjs:7]: Shouldn't the version here be 0.0.5?
```suggestion
  version: "0.0.5",
```

File: components/data_stores/actions/has-key/has-key.mjs [hunk 2]
```diff
@@ -0,0 +1,35 @@
+import app from "../../data_stores.app.mjs";
+
+export default {
+  key: "data_stores-has-key",
+  name: "Has Key",
+  description: "Check if a key exists in your [Pipedream Data Store](https://pipedream.com/data-stores/).",
+  version: "0.0.2",
```
Comment (michelle0927) [inline components/data_stores/actions/has-key/has-key.mjs:7]: This should be version 0.0.1.
```suggestion
  version: "0.0.1",
```

**Review by michelle0927 — DISMISSED** [review]

File: components/data_stores/actions/has-key/has-key.mjs [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+        }),
+      ],
+      description: "The key to check.",
+    },
+  },
+  async run ({ $ }) {
+    const hasKey = await this.dataStore.has(this.key);
+    $.export("$summary", hasKey
+      ? "Key \"" + this.key + "\" exists."
+      : "Key \"" + this.key + "\" does not exist.");
```
Comment (andrewjschuang) [inline components/data_stores/actions/has-key/has-key.mjs:32]: **Minor**

```suggestion
      ? `Key ${this.key} exists.`
      : `Key ${this.key} does not exist.`);
```

File: components/data_stores/actions/list-keys/list-keys.mjs [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+      propDefinition: [
+        app,
+        "dataStore",
+      ],
+    },
+  },
+  async run ({ $ }) {
+    const keys = await this.dataStore.keys();
+    if (keys.length > 0) {
+      $.export("$summary", "Found " + keys.length + " key(s).");
```
Comment (andrewjschuang) [inline components/data_stores/actions/list-keys/list-keys.mjs:21]: **Minor**

```suggestion
      $.export("$summary", `Found ${keys.length} key(s).`);
```

File: components/data_stores/actions/list-records/list-records.mjs [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+          });
+        }).catch((err) => {
+          reject(err);
+        });
+      }));
+    }
+
+    const records = await Promise.all(promises);
+    if (keys.length > 0) {
+      $.export("$summary", "Found " + keys.length + " record(s).");
```
Comment (andrewjschuang) [inline components/data_stores/actions/list-records/list-records.mjs:59]: **Minor**

```suggestion
      $.export("$summary", `Found ${keys.length} record(s).`);
```

**Review by andrewjschuang — APPROVED** [review]
@vellames-turing I've added just some minor comments, but LGTM for QA!

**Review by andrewjschuang — APPROVED** [review]

**Pull request merged**